src/collector/Group.cpp

```diff
@@ -334,26 +338,14 @@ void Group::update_status(bool forbidden_dht)
             if (backend_ts < cur_ts)
                 backend_ts = cur_ts;
 
-            if (b_status == Backend::BAD) {
-                have_bad = true;
-                break;
-            } else if (b_status == Backend::RO) {
+            if (b_status == Backend::RO) {
                 have_ro = true;
             } else if (b_status != Backend::OK) {
                 have_other = true;
             }
         }
```

Comment:
Yes, it is fixed in further commit: https://github.com/yandex/mastermind/commit/062327a37902b043e68cc903b1bdbb401514d4a6